cyclonedx/model/definition.py

```diff
@@ -0,0 +1,278 @@
+# This file is part of CycloneDX Python Library
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+# Copyright (c) OWASP Foundation. All Rights Reserved.
+
+from typing import TYPE_CHECKING, Any, Dict, Iterable, List, Optional, Type, Union
+from xml.etree.ElementTree import Element  # nosec B405
+
+import serializable
+from serializable.helpers import BaseHelper
+from sortedcontainers import SortedSet
+
+from .._internal.compare import ComparableTuple as _ComparableTuple
+from ..exception.serialization import CycloneDxDeserializationException
+from ..serialization import BomRefHelper
+from . import ExternalReference
+from .bom_ref import BomRef
+
+if TYPE_CHECKING:  # pragma: no cover
+    from serializable import ObjectMetadataLibrary, ViewType
+
+
+def bom_ref_or_str(bom_ref: Optional[Union[str, BomRef]]) -> BomRef:
+    if isinstance(bom_ref, BomRef):
+        return bom_ref
+    else:
+        return BomRef(value=str(bom_ref) if bom_ref else None)
+
+
+@serializable.serializable_class(serialization_types=[
+    serializable.SerializationType.JSON,
+    serializable.SerializationType.XML]
+)
+class Standard:
+    """
+    A standard of regulations, industry or organizational-specific standards, maturity models, best practices,
+    or any other requirements.
+    """
+
+    def __init__(
+        self, *,
+        bom_ref: Optional[Union[str, BomRef]] = None,
+        name: Optional[str] = None,
+        version: Optional[str] = None,
+        description: Optional[str] = None,
+        owner: Optional[str] = None,
+        # TODO: requirements: Optional[Iterable[Requirement]] = None,
+        # TODO: levels: Optional[Iterable[Level]] = None,
```

Comment:
I've left those two out because they'd have a ripple effect. I can fulfill my use-case without those two and contribute the rest in a latter PR.

Reply:
we actually do not need unused/commented-out arguments here.
instead, could you add notes about the yet left-out properties in style of https://github.com/CycloneDX/cyclonedx-python-lib/blob/2decdb73f573fda5a33a4c71c9d8b2c3b7970ee0/cyclonedx/model/bom.py#L108-L117

Reply:
Sure and done!